src/main/java/net/sf/jabref/gui/search/SearchWorker.java

```diff
@@ -60,44 +60,52 @@ protected void done() {
     }
 
     private void updateUIWithSearchResult(List<BibEntry> matchedEntries) {
+        GlobalSearchBar globalSearchBar = JabRefGUI.getMainFrame().getGlobalSearchBar();
 
         // check if still the current query
-        if (!basePanel.getSearchBar().isStillValidQuery(searchQuery)) {
+        if (!globalSearchBar.isStillValidQuery(searchQuery)) {
             // do not update - another search was already issued
             return;
         }
 
-        // clear
-        for (BibEntry entry : basePanel.getDatabase().getEntries()) {
-            entry.setSearchHit(false);
-        }
-
-        for (BibEntry entry : matchedEntries) {
-            entry.setSearchHit(true);
-        }
+        basePanel.getDatabase().getEntries().forEach(entry -> entry.setSearchHit(false));
+        matchedEntries.forEach(entry -> entry.setSearchHit(true));
 
         basePanel.getMainTable().getTableModel().updateSearchState(MainTableDataModel.DisplayOption.DISABLED);
-
         // Show the result in the chosen way:
-        switch (mode) {
-        case FLOAT:
-            basePanel.getMainTable().getTableModel().updateSearchState(MainTableDataModel.DisplayOption.FLOAT);
-            break;
-        case FILTER:
-            basePanel.getMainTable().getTableModel().updateSearchState(MainTableDataModel.DisplayOption.FILTER);
-            break;
-        default:
-            break;
+        switch (searchDisplayMode) {
+            case FLOAT:
+                basePanel.getMainTable().getTableModel().updateSearchState(MainTableDataModel.DisplayOption.FLOAT);
+                break;
+            case FILTER:
+                basePanel.getMainTable().getTableModel().updateSearchState(MainTableDataModel.DisplayOption.FILTER);
+                break;
+            default:
+                LOGGER.error("Following searchDisplayMode was not defined: " + searchDisplayMode);
+                break;
         }
 
-        // select first match (i.e., row) if there is any
-        int hits = matchedEntries.size();
-        if ((hits > 0) && (basePanel.getMainTable().getRowCount() > 0)) {
-            basePanel.getMainTable().setSelected(0);
+        // only selects the first match if the selected entries are no hits or no entry is selected
+        List<BibEntry> selectedEntries = basePanel.getSelectedEntries();
+        boolean isHitSelected = !selectedEntries.isEmpty();
+        for (BibEntry entry : selectedEntries) {
+            if (!entry.isSearchHit()) {
```

Comment:
You could simpler use streams: `isHitSelected = selectedEntries.stream().anyMatch(  condtion ) `